Sources/TuistDependencies/DependenciesController.swift

```diff
@@ -114,24 +119,30 @@ public final class DependenciesController: DependenciesControlling {
             throw DependenciesControllerError.noPlatforms
         }
 
+        var dependenciesGraph = DependenciesGraph(thirdPartyDependencies: [:])
+
         if let carthageDepedencies = dependencies.carthage, !carthageDepedencies.dependencies.isEmpty {
-            try carthageInteractor.update(
+            dependenciesGraph = try carthageInteractor.install(
                 dependenciesDirectory: dependenciesDirectory,
                 dependencies: carthageDepedencies,
-                platforms: platforms
+                platforms: platforms,
+                shouldUpdate: shouldUpdate
             )
         } else {
             try carthageInteractor.clean(dependenciesDirectory: dependenciesDirectory)
         }
 
         if let swiftPackageManagerDependencies = dependencies.swiftPackageManager, !swiftPackageManagerDependencies.packages.isEmpty {
-            try swiftPackageManagerInteractor.update(
+            try swiftPackageManagerInteractor.install(
                 dependenciesDirectory: dependenciesDirectory,
                 dependencies: swiftPackageManagerDependencies,
+                shouldUpdate: shouldUpdate,
                 swiftToolsVersion: swiftVersion
             )
         } else {
             try swiftPackageManagerInteractor.clean(dependenciesDirectory: dependenciesDirectory)
         }
+
+        try dependenciesGraphController.save(dependenciesGraph, to: path)
```

Comment:
Shouldn't this be:
```suggestion
        if dependenciesGraph.thirdPartyDependencies.isEmpty {
            try dependenciesGraphController.save(dependenciesGraph, to: path)
        } else {
            try dependenciesGraphController.clean(at: path)
        }
```
?

If instead we want to write it anyway, then we can get rid of the clean method